sdk/messaging/azservicebus/receiver.go

```diff
@@ -364,6 +361,14 @@ func (r *Receiver) receiveMessagesImpl(ctx context.Context, maxMessages int, opt
 	cancelReleaser := r.cancelReleaser.Swap(emptyCancelFn).(func() string)
 	_ = cancelReleaser()
 
+	if maxMessages <= 0 {
+		return nil, internal.NewErrNonRetriable("maxMessages should be greater than 0")
+	}
+
+	if maxMessages > int(r.maxAllowedCredits) {
```

Comment:
Does this mean that if the link has N number of credits and I call `Receive()` with N+M, `Receive()` will return an error?

Reply:
No, if you look a little below it takes into account your existing credits when calculating how many to issue.

In other words, if you ask for 10000 we'll just make sure that 10000 credits are issued, overall.

Reply:
That part I get.  I'm looking at the caller above us `Receiver.ReceiveMessages()`.  It looks to me that if I pass a value for `maxMessages` that's greater than the max link credit it returns an error.

Reply:
Talked more with @jhendrixMSFT about this and it's more interesting than I think either of us considered. We'll let this stand (for now) and then hit it again when we change how windowing+max credits works in go-amqp.  This'll protect the current implementation from possibly hanging when the underlying Receiver's channel is full.

We'd like to make a change to go-amqp to eliminate this restriction. I've filed an issue for the discussion here: https://github.com/Azure/go-amqp/issues/240